product-mini/platforms/nuttx/CMakeLists.txt

```diff
@@ -0,0 +1,173 @@
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_LOG)
+  add_definitions(-DWASM_ENABLE_LOG=1)
+else()
+  add_definitions(-DWASM_ENABLE_LOG=0)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_AOT_WORD_ALIGN_READ)
+  add_definitions(-DWASM_ENABLE_WORD_ALIGN_READ=1)
+else()
+  add_definitions(-DWASM_ENABLE_WORD_ALIGN_READ=0)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_STACK_GUARD_SIZE)
+  add_definitions(-DWASM_STACK_GUARD_SIZE=0)
+else()
+  add_definitions(
+    -DWASM_STACK_GUARD_SIZE=${CONFIG_INTERPRETERS_WAMR_STACK_GUARD_SIZE})
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_MEMORY_TRACING)
+  add_definitions(-DWASM_ENABLE_MEMORY_TRACING=1)
+else()
+  add_definitions(-DWASM_ENABLE_MEMORY_TRACING=0)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_SHARED_MEMORY)
+  set(WAMR_BUILD_SHARED_MEMORY 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_BULK_MEMORY)
+  set(WAMR_BUILD_BULK_MEMORY 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_PERF_PROFILING)
+  set(WAMR_BUILD_PERF_PROFILING 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_MEMORY_PROFILING)
+  set(WAMR_BUILD_MEMORY_PROFILING 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_MULTI_MODULE)
+  set(WAMR_BUILD_MULTI_MODULE 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_LIB_PTHREAD_SEMAPHORE)
+  set(WAMR_BUILD_LIB_PTHREAD_SEMAPHORE 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_DISABLE_HW_BOUND_CHECK)
+  set(WAMR_DISABLE_HW_BOUND_CHECK 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_CUSTOM_NAME_SECTIONS)
+  set(WAMR_BUILD_CUSTOM_NAME_SECTION 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_GLOBAL_HEAP_POOL)
+  set(WAMR_BUILD_GLOBAL_HEAP_POOL 1)
+  math(EXPR _HEAP_SIZE_
+       "${CONFIG_INTERPRETERS_WAMR_GLOBAL_HEAP_POOL_SIZE} * 1024")
+  set(WAMR_BUILD_GLOBAL_HEAP_SIZE ${_HEAP_SIZE_})
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_ENABLE_SPEC_TEST)
+  set(WAMR_BUILD_SPEC_TEST 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_REF_TYPES)
+  set(WAMR_BUILD_REF_TYPES 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_CLASSIC)
+  # include iwasm_interp.cmake
+  set(WAMR_BUILD_INTERP 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_FAST)
+  # enable iwasm_interp.cmake
+  set(WAMR_BUILD_FAST_INTERP 1)
+endif()
+
+if((CONFIG_INTERPRETERS_WAMR_FAST OR CONFIG_INTERPRETERS_WAMR_CLASSIC)
+   AND CONFIG_INTERPRETERS_WAMR_MINILOADER)
+  # enable iwasm_interp.cmake
+  set(WAMR_BUILD_MINI_LOADER 1)
+else()
+  set(WAMR_BUILD_MINI_LOADER 0)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_AOT)
+  # include iwasm_aot.cmake
+  set(WAMR_BUILD_AOT 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_DEBUG_INTERP)
+  # include debug_engine.cmake
+  set(WAMR_BUILD_DEBUG_INTERP 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_LIBC_BUILTIN)
+  # include libc_builtin.cmake
+  set(WAMR_BUILD_LIBC_BUILTIN 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_LIBC_WASI)
+  # include libc_wasi.cmake
+  set(WAMR_BUILD_LIBC_WASI 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_THREAD_MGR)
+  # include thread_mgr.cmake
+  set(WAMR_BUILD_THREAD_MGR 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_LIB_PTHREAD)
+  # include lib_pthread.cmake
+  set(WAMR_BUILD_LIB_PTHREAD 1)
+endif()
```

Comment:
Missing WAMR_BUILD_GC and WAMR_BUILD_STRINGREF

Reply:
> Missing WAMR_BUILD_GC and WAMR_BUILD_STRINGREF

I added WAMR_BUILD_GC in the latest patchset, but I can't seem to find where to set WAMR_BUILD_STRINGREF

Reply:
It's in `build-scripts/config_common.cmake:348`

Reply:
> WAMR_BUILD_STRINGREF

😁 thx for you help, i added it to the latest patchset
```cmake
if(CONFIG_INTERPRETERS_WAMR_GC)
  set(WAMR_BUILD_GC 1)
  set(WAMR_BUILD_STRINGREF 1)
  set(WAMR_BUILD_REF_TYPES 1)
endif()
```